src/Campaigns/Controllers/CampaignRequestController.php

```diff
@@ -0,0 +1,159 @@
+
+        $base = add_query_arg(
+            map_deep($request->get_query_params(), function ($value) {
+                if (is_bool($value)) {
+                    $value = $value ? 'true' : 'false';
+                }
+
+                return urlencode($value);
+            }),
+            rest_url(CampaignRoute::CAMPAIGNS)
+        );
+
+        if ($page > 1) {
+            $prevPage = $page - 1;
+
+            if ($prevPage > $totalPages) {
+                $prevPage = $totalPages;
+            }
+
+            $response->link_header('prev', add_query_arg('page', $prevPage, $base));
+        }
+
+        if ($totalPages > $page) {
+            $nextPage = $page + 1;
+            $response->link_header('next', add_query_arg('page', $nextPage, $base));
+        }
+
+        return $response;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @return WP_Error | WP_REST_Response
+     *
+     * @throws Exception
+     */
+    public function updateCampaign(WP_REST_Request $request)
+    {
+        $campaign = Campaign::find($request->get_param('id'));
+
+        if ( ! $campaign) {
+            return new WP_Error('campaign_not_found', __('Campaign not found', 'give'), ['status' => 404]);
+        }
+
+        $statusMap = [
+            'draft' => CampaignStatus::DRAFT(),
+            'active' => CampaignStatus::ACTIVE(),
+        ];
+
+        foreach ($request->get_params() as $key => $value) {
+            switch ($key) {
+                case 'id':
+                    break;
+                case 'status':
+                    $status = array_key_exists($value, $statusMap)
+                        ? $statusMap[$value]
+                        : CampaignStatus::DRAFT();
+
+                    $campaign->setAttribute('status', $status);
+
+                    break;
+                default:
+                    if ($campaign->hasProperty($key)) {
+                        $campaign->setAttribute($key, $value);
+                    }
+            }
+        }
+
+        if ($campaign->isDirty()) {
+            $campaign->save();
+        }
+
+        return new WP_REST_Response($campaign->toArray());
+    }
+
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function createCampaign(WP_REST_Request $request): WP_REST_Response
+    {
+        $campaign = Campaign::create([
+            'type' => CampaignType::CORE(),
+            'title' => $request->get_param('title'),
+            'shortDescription' => $request->get_param('shortDescription'),
+            'longDescription' => '',
+            'logo' => '',
+            'image' => '',
+            'primaryColor' => '',
+            'secondaryColor' => '',
+            'goal' => 1000,
+            'goalType' => 'amount',
+            'status' => CampaignStatus::DRAFT(),
+            'startDate' => $request->get_param('startDateTime'),
+            'endDate' => $request->get_param('endDateTime'),
```

Comment:
As these fields are not required on the UI, I think we can replace it by this:

```
'startDate' => $request->get_param('startDateTime') ?? null,
'endDate' => $request->get_param('endDateTime') ?? null,
```

Reply:
If any of these parameters are not set, `get_param` will return null anyway because there is no default value set.